servers/rendering/storage/render_data.cpp

```diff
@@ -35,15 +35,23 @@ void RenderData::_bind_methods() {
 	ClassDB::bind_method(D_METHOD("get_render_scene_data"), &RenderData::get_render_scene_data);
 	ClassDB::bind_method(D_METHOD("get_environment"), &RenderData::get_environment);
 	ClassDB::bind_method(D_METHOD("get_camera_attributes"), &RenderData::get_camera_attributes);
+	ClassDB::bind_method(D_METHOD("get_light_storage"), &RenderData::get_light_storage);
 }
 
 void RenderDataExtension::_bind_methods() {
 	GDVIRTUAL_BIND(_get_render_scene_buffers);
 	GDVIRTUAL_BIND(_get_render_scene_data)
+	GDVIRTUAL_BIND(_get_light_storage)
 	GDVIRTUAL_BIND(_get_environment)
 	GDVIRTUAL_BIND(_get_camera_attributes)
 }
 
+RendererLightStorage *RenderDataExtension::get_light_storage() const {
+	RendererLightStorage *ret = nullptr;
```

Comment:
So light storage, as with the other storage classes, are part of the core rendering system. You can just return this through `https://github.com/godotengine/godot/blob/master/servers/rendering/rendering_server_globals.cpp`.
It can then be cast to (`RendererRD::`)`LightStorage` in GDScript.

Reply:
@BastiaanOlij I followed your advice and got rid of the LightStorage class usage in RenderDataRD class, I also removed the implementation of the method from there and added it in the base class using RenderingServerGlobals instead. But when you say it doesnt need to be virtual, I guess it should stay virtual because at least RenderDataExtension class is inherited from RenderData and if Im not mistaken it sould be able to override the method in order to add custom behaviour, right?